wrong-exception-operation raised when concat tuple in except

### Bug description

Pylint wrong-exception-operation is raised when concatenating the tuple in except. 

Consider this file `a.py`:

```python
EXC = (ValueError,)

try:
	int("hi")
except (OSError,) + EXC as exc:
	print("caught it (exc={})".format(exc))

try:
	raise OSError("whatever")
except (OSError,) + EXC as exc:
	print("caught it (exc={})".format(exc))
```

Pylint complains with:

```
************* Module a
a.py:5:7: W0716: Invalid exception operation. Did you mean '((OSError, ), EXC)' instead? (wrong-exception-operation)
a.py:10:7: W0716: Invalid exception operation. Did you mean '((OSError, ), EXC)' instead? (wrong-exception-operation)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

when this case is valid.


### Configuration

_No response_

### Command used

```shell
pylint a.py
```


### Pylint output

```python
************* Module a
a.py:5:7: W0716: Invalid exception operation. Did you mean '((OSError, ), EXC)' instead? (wrong-exception-operation)
a.py:10:7: W0716: Invalid exception operation. Did you mean '((OSError, ), EXC)' instead? (wrong-exception-operation)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```


### Expected behavior

`wrong-exception-operation` shouldn't be raised

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/9288.false_positive
+++ b/doc/whatsnew/fragments/9288.false_positive
@@ -1,0 +1,4 @@
+Fix false positive for ``invalid-exception-operation`` when concatenating tuples
+of exception types.
+
+Closes #9288

--- a/pylint/checkers/exceptions.py
+++ b/pylint/checkers/exceptions.py
@@ -533,8 +533,19 @@ class ExceptionsChecker(checkers.BaseChecker):
     @utils.only_required_for_messages("wrong-exception-operation")
     def visit_binop(self, node: nodes.BinOp) -> None:
         if isinstance(node.parent, nodes.ExceptHandler):
+            both_sides_tuple_or_uninferable = isinstance(
+                utils.safe_infer(node.left), (nodes.Tuple, util.UninferableBase)
+            ) and isinstance(
+                utils.safe_infer(node.right), (nodes.Tuple, util.UninferableBase)
+            )
+            # Tuple concatenation allowed
+            if both_sides_tuple_or_uninferable:
+                if node.op == "+":
+                    return
+                suggestion = f"Did you mean '({node.left.as_string()} + {node.right.as_string()})' instead?"
             # except (V | A)
-            suggestion = f"Did you mean '({node.left.as_string()}, {node.right.as_string()})' instead?"
+            else:
+                suggestion = f"Did you mean '({node.left.as_string()}, {node.right.as_string()})' instead?"
             self.add_message("wrong-exception-operation", node=node, args=(suggestion,))
 
     @utils.only_required_for_messages("wrong-exception-operation")

--- a/tests/functional/w/wrong_exception_operation.py
+++ b/tests/functional/w/wrong_exception_operation.py
@@ -16,3 +16,24 @@ try:
     1/0
 except (ValueError < TypeError): # [wrong-exception-operation]
     pass
+
+
+# Concatenation of exception type tuples
+DIVISION_BY_ZERO = (ZeroDivisionError,)
+VALUE_ERROR = (ValueError,)
+UNINFERABLE = DIVISION_BY_ZERO | VALUE_ERROR
+
+try:
+    1/0
+except (ValueError, ) + DIVISION_BY_ZERO:
+    pass
+
+try:
+    1/0
+except (ValueError, ) | DIVISION_BY_ZERO:  # [wrong-exception-operation]
+    pass
+
+try:
+    1/0
+except (ValueError, ) + UNINFERABLE:
+    pass

--- a/tests/functional/w/wrong_exception_operation.txt
+++ b/tests/functional/w/wrong_exception_operation.txt
@@ -2,3 +2,4 @@ catching-non-exception:6:8:6:30::"Catching an exception which doesn't inherit fr
 wrong-exception-operation:6:8:6:30::Invalid exception operation. Did you mean '(ValueError, TypeError)' instead?:UNDEFINED
 wrong-exception-operation:11:8:11:30::Invalid exception operation. Did you mean '(ValueError, TypeError)' instead?:UNDEFINED
 wrong-exception-operation:17:8:17:30::Invalid exception operation. Did you mean '(ValueError, TypeError)' instead?:UNDEFINED
+wrong-exception-operation:33:7:33:40::Invalid exception operation. Did you mean '((ValueError, ) + DIVISION_BY_ZERO)' instead?:UNDEFINED

